eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -134,6 +134,13 @@ class Setup(val datadir: File,
   }
 
   val nodeParams = NodeParams.makeNodeParams(config, instanceId, nodeKeyManager, channelKeyManager, initTor(), databases, blockHeight, feeEstimator, pluginParams)
+
+  // validate that channel seed is the same that was used when local channels were created
+  nodeParams.db.channels.listLocalChannels().foreach(hasCommitments => Try(Commitments.validateSeed(hasCommitments.commitments, channelKeyManager)) match {
+    case Success(_) => ()
+    case Failure(e) => throw InvalidChannelSeedException(e)
+  })
```

Comment:
This should be done here imo: https://github.com/ACINQ/eclair/blob/36ed788d392e53a95f707c33dd0930871c72936a/eclair-core/src/main/scala/fr/acinq/eclair/io/Switchboard.scala#L46-L60

Main reasons:
- single place where we load and check channels
- saves one (expensive) call to `listLocalChannels()`
- referencing `Commitments` from `Setup` feels weird.

Reply:
I chose to add the check in `Setup` to make sure that it's the very first thing we do with channel data. If the current channel seed does not match a channel's public key it means that something is very wrong and we should exit immediately. 

Once setup is over I think that we use `listLocalChannels()` in different places, in the following order:

1. DBCompatChecker
2. PostRestartHtlcCleaner
3. Switchboard

For this to be safe we have to make sure that everything we do before `Switchboard` starts is safe even if the channel seed has been modified which feels hard to enforce ?

Reply:
Or we could add the seed check to `DBCompatChecker` ?

Reply:
> Or we could add the seed check to DBCompatChecker ?

That would be a good choice, I agree with you that it's good to do this at the beginning of `Setup`, before we spawn actors.
We should probably rename `DbCompatChecker` then to test both DB + seed, something like `StartupCompatChecker`?

Reply:
>> Or we could add the seed check to DBCompatChecker ?

> That would be a good choice

+1

> We should probably rename DbCompatChecker then to test both DB + seed, something like StartupCompatChecker?

Or `DbChecker`?